src/CalcManager/CEngine/CalcInput.cpp

```diff
@@ -269,34 +269,47 @@ bool CalcInput::IsEmpty()
 wstring CalcInput::ToString(uint32_t radix)
 {
     // In theory both the base and exponent could be C_NUM_MAX_DIGITS long.
-    wstringstream resStream;
-
     if ((m_base.value.size() > MAX_STRLEN) || (m_hasExponent && m_exponent.value.size() > MAX_STRLEN))
     {
         return wstring();
     }
 
+    wstring result;
+
     if (m_base.IsNegative())
     {
-        resStream << L'-';
+        result += L'-';
```

Comment:
`=` is ~15% faster than `+=`, let's use it here instead (as well as line 286 and 290)

Reply:
I think we can use `=` for this line, but I don't believe we can use it for lines 286 and 290 because that would overwrite the value that is in result which might contain a `-` if IsNegative is true.

Reply:
right!